app/src/main/java/org/horaapps/leafpic/activities/MainActivity.java

```diff
@@ -168,21 +168,8 @@ public void displayMedia(Album album) {
     @Override
     public void onMediaClick(Album album, ArrayList<Media> media, int position) {
 
-        if (!pickMode) {
-            Intent intent = new Intent(getApplicationContext(), SingleMediaActivity.class);
-            intent.putExtra("album", album);
-            try {
-                intent.setAction(SingleMediaActivity.ACTION_OPEN_ALBUM);
-                intent.putExtra("media", media);
-                intent.putExtra("position", position);
-                startActivity(intent);
-            } catch (Exception e) {
-                intent.setAction(SingleMediaActivity.ACTION_OPEN_ALBUM_LAZY);
```

Comment:
this is necessary when there are a lot of media in an album, the Bundle has a size limit.